rs/sns/governance/src/governance/assorted_governance_tests.rs

```diff
@@ -3330,6 +3330,75 @@ fn test_upgrade_proposals_blocked_by_pending_upgrade() {
     }
 }
 
+#[test]
+fn test_upgrade_proposals_not_blocked_by_old_proposals_upgrade() {
+    // Step 1: Prepare the world.
+    use ProposalDecisionStatus as Status;
+
+    let upgrade_action_id: u64 =
+        (&Action::UpgradeSnsControlledCanister(UpgradeSnsControlledCanister::default())).into();
+
+    let proposal_id = 1_u64;
+    let some_other_proposal_id = 99_u64;
+    let proposal = ProposalData {
+        action: upgrade_action_id,
+        id: Some(proposal_id.into()),
+        decided_timestamp_seconds: NativeEnvironment::DEFAULT_TEST_START_TIMESTAMP_SECONDS
```

Comment:
Could you summarize in the doc string what this test is checking, please?

Reply:
Done!